ldap/servers/slapd/back-ldbm/archive.c

```diff
@@ -490,3 +490,212 @@ ldbm_back_ldbm2archive(Slapi_PBlock *pb)
+                    source, prerr, slapd_pr_strerror(prerr));
+        }
+        slapi_log_err(SLAPI_LOG_ERR, "archive_copyfile",
+                      "Source file (%s) could not be accessed - error %d (%s)\n",
+                      source, prerr, slapd_pr_strerror(prerr));
+        return -1;
+    }
+
+    /* malloc the buffer */
+    buffer = slapi_ch_malloc(64 * 1024);
+
+    /* Open source file */
+    source_fd = open(source, O_RDONLY, 0);
+    if (source_fd == -1) {
+        PRErrorCode prerr = PR_GetError();
+        if (task) {
+            slapi_task_log_notice(task,
+                    "archive_copyfile - Source file (%s) could not be opened - error %d (%s)",
+                    source, prerr, slapd_pr_strerror(prerr));
+        }
+        slapi_log_err(SLAPI_LOG_ERR, "archive_copyfile",
+                      "Source file (%s) could not be opened - error %d (%s)\n",
+                      source, prerr, slapd_pr_strerror(prerr));
+        goto error;
+    }
+
+    /* Open destination file */
+    dest_file = slapi_ch_smprintf("%s/%s", destination, filename);
+    dest_fd = open(dest_file, O_CREAT | O_WRONLY, mode);
+    if (dest_fd == -1) {
+        PRErrorCode prerr = PR_GetError();
+        if (task) {
+            slapi_task_log_notice(task,
+                    "archive_copyfile - Destination file (%s) could not be opened - error %d (%s)",
+                    destination, prerr, slapd_pr_strerror(prerr));
+        }
+        slapi_log_err(SLAPI_LOG_ERR, "archive_copyfile",
+                      "Destination file (%s) could not be opened - error %d (%s)\n",
+                      dest_file, prerr, slapd_pr_strerror(prerr));
+        goto error;
+    }
+
+    slapi_log_err(SLAPI_LOG_INFO, "archive_copyfile",
+                  "Copying %s to %s\n", source, dest_file);
+    if (task) {
+        slapi_task_log_notice(task,
+                "archive_copyfile - Copying %s to %s",
+                source, dest_file);
+    }
+
+    /* Loop round reading data and writing it */
+    while (1) {
+        int i;
+        char *ptr = NULL;
+        return_value = read(source_fd, buffer, 64 * 1024);
+        if (return_value <= 0) {
+            /* means error or EOF */
+            if (return_value < 0) {
+                PRErrorCode prerr = PR_GetError();
+                if (task) {
+                    slapi_task_log_notice(task,
+                            "archive_copyfile - Failed to read (%s) error %d (%s) - rc %d",
+                            source, prerr, slapd_pr_strerror(prerr), return_value);
+                }
+                slapi_log_err(SLAPI_LOG_ERR, "archive_copyfile",
+                              "Failed to read (%s) error %d (%s) - rc %d\n",
+                            source, prerr, slapd_pr_strerror(prerr), return_value);
+            }
+            break;
+        }
+        bytes_to_write = return_value;
+        ptr = buffer;
+        for (i = 0; i < CPRETRY; i++) {
```

Comment:
minor, inline the size_t i here.

Reply:
:-) copy and paste, on it...

Reply:
Not sure, you are using 'i' for the success of the retry count.